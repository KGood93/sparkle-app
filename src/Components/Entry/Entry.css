.entry {
    display: flex;
    flex-direction: column;
    justify-content: center;

}

.home {
    display: block;
    margin: auto;
}

.name {
    display: flex;
    flex-direction: row;
    margin-top: 15px;
    justify-content: center;
}

.nameLabel {
    margin-right: 5px;
    font-size: 25px;
    font-family: 'Playfair Display', serif;
    color: rgb(6,106,117);
    font-weight: bold;

}

.titleInput {
    width: 175px;
    height: 30px;
    border: solid rgb(6,106,117) 2px;
    font-family: 'Playfair Display', serif;
    font-size: 14pt;
    padding: 2px;
    padding-left: 4px;
}

.eContent {
    display: block;
    margin-left: auto;
    margin-right: auto;
    margin-top: 10px;
    width: 275px;
    border: 3px solid rgb(6,106,117);
    font-family: 'Playfair Display', serif;
    font-size: 14pt;
}

.deleteEntry {
    display: block;
    margin: auto;
    margin-bottom: 15px;
    border: solid rgb(6,106,117) 2px;
    font-family: 'Playfair Display', serif;
    font-weight: bold;
    font-size: 20px;
    height: 34px;
    color: rgb(6,106,117);
    background-color: rgb(241,221,210);
}