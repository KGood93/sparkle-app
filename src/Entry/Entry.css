.homeNav {
    display: flex;
    justify-content: center;    
}

.entryMain {
    display: flex;
    flex-direction: column;
    align-items: center;
}

.motivQuote {
    border: 3px solid black;
    margin: 15px;
    margin-bottom: 0px;
    width: 350px;
}

.motivQuote h2 {
    margin: 5px;
    text-align: center;
}

.motivQuote h4 {
    margin: 5px;
    text-align: center;
}

textarea {
    height: 250px;
    width: 350px;
    margin-top: 15px;
    margin-bottom: 15px;
}

.entryButtons{
    display: flex;
    flex-direction: row;
    justify-content: center;
    margin-bottom: 100px;
}

.buttons {
    align-content: flex-end;
}