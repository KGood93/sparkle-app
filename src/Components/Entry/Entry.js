//fetches and displays an already existing entry//

import React from 'react'
import {format} from 'date-fns'
import './Entry.css'
import { allEntries } from '../../STORE'
import Quote from '../Quote/Quote'

class Entry extends React.Component {

    handleClickDelete = e => {
        e.preventDefault()
    }
    
    render() {
        console.log(this.props.title) 
        return (
            <div className='entry'>
                <h2 className='entryTitle'>
                    {this.props.title}
                </h2>
                <Quote />
                <p>
                    {this.props.content}
                </p>
                <button className='deleteEntry' type='button'>
                    {' '}
                    Remove
                </button>

            </div>
        )
    }
}

export default Entry