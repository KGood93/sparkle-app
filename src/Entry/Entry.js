import React from 'react';
import './Entry.css';

class Entry extends React.Component {

    render() {
        return (
          <div className="main">
            <div className="entryMain">
              <section className="motivQuote">
                <h2>Motivation Quote Prompt</h2>
                <h4 className="quoteBy">Quote Name</h4>
              </section>
      
              <section>
                <textarea></textarea>
              </section>
            </div>
      
            <div className="entryButtons">
              <section className="buttons">
                  <button type="button" className="clear">Clear Entry</button>
                  <button type="button" className="edit">Edit Entry</button>
                  <button type="button" className="save">Save Entry</button>
              </section>
            </div>
          </div>
        )
    }
}
export default Entry;