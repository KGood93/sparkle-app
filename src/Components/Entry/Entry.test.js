import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import Entry from './Entry';

describe('Entry Component', () => {
    it(`renders without crashing`, () => {
        const wrapper = shallow(<Entry />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});