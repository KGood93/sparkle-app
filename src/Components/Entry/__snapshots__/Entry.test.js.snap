// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Entry Component renders without crashing 1`] = `
<div
  className="entry"
>
  <div
    className="home"
  >
    <Link
      to="/journal"
    >
      Home
    </Link>
  </div>
  <div
    className="name"
  >
    <label
      className="nameLabel"
      htmlFor="EntryName"
    >
      Title:
    </label>
    <input
      className="titleInput"
      id="entryName"
      name="entryName"
      type="text"
    />
  </div>
  <Quote />
  <div
    className="entryContent"
  >
    <textarea
      className="eContent"
      id="entryContent"
      name="entryContent"
    />
  </div>
</div>
`;
